File: common/src/main/java/cn/rubintry/common/model/BaseModel.java
package cn.rubintry.common.model;

import cn.rubintry.common.utils.http.IMonitor;

/**
 * @author logcat
 */
public class BaseModel<T> implements IMonitor {
    private T data;
    private int errorCode;
    private String errorMsg;

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }



    @Override
    public String getErrMsg() {
        return errorMsg;
    }

    @Override
    public int getCode() {
        return errorCode;
    }

    @Override
    public boolean success() {
        return errorCode == 0;
    }
}

File: common/src/main/java/cn/rubintry/common/model/BaseModel.kt
package cn.rubintry.common.model

import cn.rubintry.common.utils.http.IMonitor

/**
 * @author logcat
 */
class BaseModel<T> : IMonitor {
    override val errMsg: String
        get() = errorMsg
    override val code: Int
        get() = errorCode

    var data: T? = null
        get() = field
        private set


    var errorMsg: String = ""
        get() = field
        set

    var errorCode: Int = 0
        get() = field
        set


    fun setData(data: T) {
        this.data = data
    }


    override fun success(): Boolean {
        return code == 0
    }
}